homeassistant/components/litterrobot/button.py

```diff
@@ -23,22 +26,64 @@ async def async_setup_entry(
 ) -> None:
     """Set up Litter-Robot cleaner using config entry."""
     hub: LitterRobotHub = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities(
-        LitterRobotResetWasteDrawerButton(
-            robot=robot, entity_type=TYPE_RESET_WASTE_DRAWER, hub=hub
-        )
-        for robot in hub.litter_robots()
-        if isinstance(robot, LitterRobot3)
+    entities: Iterable[LitterRobotButtonEntity] = itertools.chain(
+        (
+            LitterRobotButtonEntity(
+                robot=robot, hub=hub, description=LITTER_ROBOT_BUTTON
+            )
+            for robot in hub.litter_robots()
+            if isinstance(robot, LitterRobot3)
+        ),
+        (
+            LitterRobotButtonEntity(
+                robot=robot, hub=hub, description=FEEDER_ROBOT_BUTTON
+            )
+            for robot in hub.feeder_robots()
+        ),
     )
+    async_add_entities(entities)
+
+
+@dataclass
+class RobotButtonEntityDescription(ButtonEntityDescription, Generic[_RobotT]):
+    """A class that describes robot button entities."""
+
+    press_fn: Callable[[_RobotT], Coroutine[Any, Any, bool] | None] = lambda _: None
+
+
```

Comment:
I think press_fn should not have a default value. Instead you should create a Mixin class.
Something like this...

```suggestion
@dataclass
class RobotButtonEntityDescriptionMixin(Generic[_RobotT]):
    """A class that describes robot button entities."""

    press_fn: Callable[[_RobotT], Coroutine[Any, Any, bool] | None] = lambda _: None


@dataclass
class RobotButtonEntityDescription(ButtonEntityDescription, RobotButtonEntityDescriptionMixin[_RobotT]):
    """A class that describes robot button entities."""

```